tests/util/test_config.py

```diff
@@ -0,0 +1,151 @@
+            "farmer_peers": [
+                {
+                    "host": "127.0.0.1",
+                    "port": 8447,
+                },
+                {
+                    "host": "my.other.farmer.tld",
+                    "port": 18447,
+                },
+            ],
+        },
+        requested_node_type=NodeType.FARMER,
+        expected_peer_infos={
+            UnresolvedPeerInfo(host="my.farmer.tld", port=uint16(18447)),
+            UnresolvedPeerInfo(host="127.0.0.1", port=uint16(8447)),
+            UnresolvedPeerInfo(host="my.other.farmer.tld", port=uint16(18447)),
+        },
+    ),
+    GetUnresolvedPeerInfosCase(
+        description="multiple full node peers",
+        service_config={
+            "full_node_peers": [
+                {
+                    "host": "127.0.0.1",
+                    "port": 8444,
+                },
+                {
+                    "host": "my.full-node.tld",
+                    "port": 18444,
+                },
+            ],
+        },
+        requested_node_type=NodeType.FULL_NODE,
+        expected_peer_infos={
+            UnresolvedPeerInfo(host="127.0.0.1", port=uint16(8444)),
+            UnresolvedPeerInfo(host="my.full-node.tld", port=uint16(18444)),
+        },
+    ),
+    GetUnresolvedPeerInfosCase(
+        description="single full node peer",
+        service_config={
+            "full_node_peer": {
+                "host": "my.full-node.tld",
+                "port": 18444,
+            },
+        },
+        requested_node_type=NodeType.FULL_NODE,
+        expected_peer_infos={
+            UnresolvedPeerInfo(host="my.full-node.tld", port=uint16(18444)),
+        },
+    ),
+    GetUnresolvedPeerInfosCase(
+        description="single full node peer and multiple full node peers",
+        service_config={
+            "full_node_peer": {
+                "host": "my.full-node.tld",
+                "port": 18444,
+            },
+            "full_node_peers": [
+                {
+                    "host": "127.0.0.1",
+                    "port": 8444,
+                },
+                {
+                    "host": "my.other.full-node.tld",
+                    "port": 18444,
+                },
+            ],
+        },
+        requested_node_type=NodeType.FULL_NODE,
+        expected_peer_infos={
+            UnresolvedPeerInfo(host="my.full-node.tld", port=uint16(18444)),
+            UnresolvedPeerInfo(host="127.0.0.1", port=uint16(8444)),
+            UnresolvedPeerInfo(host="my.other.full-node.tld", port=uint16(18444)),
+        },
+    ),
+    GetUnresolvedPeerInfosCase(
+        description="no peer info in config",
+        service_config={},
+        requested_node_type=NodeType.FULL_NODE,
+        expected_peer_infos=set(),
+    ),
+)
+def test_get_unresolved_peer_infos(case: GetUnresolvedPeerInfosCase) -> None:
+    assert case.expected_peer_infos == get_unresolved_peer_infos(case.service_config, case.requested_node_type)
```

Comment:
While it doesn't affect the test passing or failing, pytest does have some 'expectation' that the expected value is on the right side of the equality check.  It uses specific words in the output I think that would align better that way.

```suggestion
    assert get_unresolved_peer_infos(case.service_config, case.requested_node_type) == case.expected_peer_infos
```